### PR TITLE
edge: make amqp error checking better

### PR DESCRIPTION
**Pull request by hkseo95**
- get rid of a function that don't need to be separated

### CODE REVIEW

File: chamge/types.h [hunk 1]
```diff
@@ -23,6 +23,12 @@ typedef enum {
   CHAMGE_RETURN_ASYNC,
 } ChamgeReturn;
 
+typedef enum {
+  CHAMGE_ERROR_AMQP = 1,
+  CHAMGE_ERROR_NODE,
+  CHAMGE_ERROR_BACKEND,
+} ChamgeError;
```
Comment (justinjoy) [inline chamge/types.h]: According to the above declaration of errors, this enum can be;

```suggestion:-4+0
typedef enum {
  CHAMGE_BACKEND_ERROR_INVALID_PARAMETER,
  CHAMGE_BACKEND_ERROR_INACCASSIBLE,
  CHAMGE_BACKEND_ERROR_OPERATION_FAILURE,
} ChamgeBackendError;
```
But, the naming is up to you.
Surely, declaring quarks should be in a separated commit.

File: chamge/types.c [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/**
+ *  Copyright 2019 SK Telecom, Co., Ltd.
+ *    Author: Jeongseok Kim <jeongseok.kim@sk.com>
```
Comment (justinjoy) [inline chamge/types.c]: That's not me for this file. :)

And please split this commit into two;  *add backend error quark*, and *edge: make amqp error checking better*.

The latter depends on the former, but their topics are different.

**Comment by hkseo95** [comment]
Do we need to fix "Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126)." checking error?

**Comment by justinjoy** [comment]
> Do we need to fix "Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126)." checking error?

No, the report from codacy is just a reference, not mandatory.

**Pull request merged**